[core IR] Add a core decomposition for aten.all

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110093

## Context

Change the ref implementation of `aten.all` to only use other `torch` operators such that we can use it for the core ATen decomposition table. This will replace the decomposition for `aten.all` that was used specifically by Inductor.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

SS-JIA committed:
[core IR] Add a core decomposition for aten.all

[ghstack-poisoned]

lezcano commented on torch/_refs/__init__.py:
```diff
@@ -2224,20 +2224,20 @@ def _fn(*args, out=None, **kwargs):
+    # torch.any and torch.all both do not currently support torch.any(x, keepdim=True), thus the dim=None
+    # case must be handled separately.
+    if dim is None:
+        result = torch.logical_not(torch.any(torch.logical_not(a)))
```
Could you add an `assert not keepdim` here to be safe?

SS-JIA commented:
```
PYTORCH_TEST_WITH_DYNAMO=1 python test_ops.py -k test_out_warning__refs_clamp_cpu
```

Test failure also reproducible on base commit; probably unrelated to these changes.

SS-JIA committed:
Update on "[core IR] Add a core decomposition for aten.all"


## Context

Change the ref implementation of `aten.all` to only use other `torch` operators such that we can use it for the core ATen decomposition table. This will replace the decomposition for `aten.all` that was used specifically by Inductor.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

SS-JIA commented:
@pytorchbot merge -i
